toolbox/deps_update/githubClient.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+var (
+	tokenFile = flag.String("token_file", "", "File containing Auth Token.")
+	owner     = flag.String("owner", "istio-testing", "Github Owner or org.")
+	token     = ""
+)
+
+type githubClient struct {
+	client *github.Client
+}
+
+// Get github api token from tokenFile
+func getToken() (string, error) {
```

Comment:
is the token file expected to exist at the beginning of the program execution? is it not expected to change?

if so...maybe read it in main, and then set it...or use a golang `init()` function? then make the program crash if it doesn't exist and then you can avoid having to ever return an error for get token and instead of getToken you can just reference the token variable. or put it on a struct somewhere...